src/cardanoLauncher.ts

```diff
@@ -230,18 +230,22 @@ export class Launcher {
    * server is ready to accept requests.
    */
   start(): Promise<Api> {
-    this.nodeService.start();
-    this.walletService.start();
-
     const stopWaiting = () =>
       this.nodeService.getStatus() > ServiceStatus.Started ||
       this.walletService.getStatus() > ServiceStatus.Started;
 
-    this.waitForApi(stopWaiting).then(() => {
-      this.walletBackend.events.emit('ready', this.walletBackend.getApi());
-    });
-
     return new Promise((resolve, reject) => {
+      this.nodeService.start();
+      this.walletService.start();
+
+      this.waitForApi(stopWaiting)
+        .then(() => {
+          this.walletBackend.events.emit('ready', this.walletBackend.getApi());
+        })
+        .catch(_ => {
+          // stopWaiting has returned true -- ignore
+        });
```

Comment:
Shouldn't this reject the promise, or is the 'exit' event expected to be emitted no matter what?